File: week1/Bu$ra/ProductofArrayExceptSelf.cs
public class Solution {
    public static int[] ProductExceptSelf(int[] nums) {
        int n = nums.Length;

       
        int[] leftProducts = new int[n];
        leftProducts[0] = 1;
        for (int i = 1; i < n; i++) {
            leftProducts[i] = leftProducts[i - 1] * nums[i - 1];
        }

        
        int rightProduct = 1;

      
        int[] result = new int[n];

        
        for (int i = n - 1; i >= 0; i--) {
            result[i] = leftProducts[i] * rightProduct;
            rightProduct *= nums[i];
        }

        return result;
    }

File: week1/esadmsk/product-of-array-except-self.py
class Solution(object):
    def productExceptSelf(self, nums):
        n = len(nums)
        left_products = [1] * n
        right_products = [1] * n
        result = [1] * n

        for i in range(1, n):
            left_products[i] = left_products[i - 1] * nums[i - 1]

        for i in range(n - 2, -1, -1):
            right_products[i] = right_products[i + 1] * nums[i + 1]

        for i in range(n):
            result[i] = left_products[i] * right_products[i]

        return result


